src/task_controller.cpp

```diff
@@ -342,7 +387,8 @@ bool MyTCServer::on_value_command(std::shared_ptr<isobus::ControlFunction> partn
 
 		case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualWorkState):
 		{
-			clients[partner].set_setpoint_work_state(processDataValue == 1);
+			// Store the work state per element rather than globally
```

Comment:
This is a good change, but you should then do it everywhere? Same for `actualWorkState`, make that also per element